### PR TITLE
Add blob nonce and endpoint_config_connection_success test

### PR DESCRIPTION
**Pull request by johnrandolph**

### CODE REVIEW

File: pubber/src/main/java/daq/pubber/Pubber.java [hunk 1]
```diff
@@ -837,6 +837,13 @@ private void extractEndpointBlobConfig() {
       String iotConfig = extractConfigBlob(IOT_ENDPOINT_CONFIG.value());
       extractedEndpoint = iotConfig == null ? null
           : OBJECT_MAPPER.readValue(iotConfig, EndpointConfiguration.class);
+      if (extractedEndpoint != null) {
+        // TODO: Refactor extractConfigBlob() to get any blob meta parameters like nonce.
+        if (deviceConfig.blobset.blobs.get(IOT_ENDPOINT_CONFIG.value()) != null) {
```
Comment (grafnu) [inline pubber/src/main/java/daq/pubber/Pubber.java:842]: nit: there's a .containsKey() method that's preferred (according to "them") over the .get() != null check...

File: schema/configuration_endpoint.json [hunk 2]
```diff
@@ -20,6 +20,9 @@
     },
     "client_id": {
       "type": "string"
+    },
+    "nonce": {
+      "type": "string"
```
Comment (grafnu) [inline schema/configuration_endpoint.json:25]: matching pattern for nonce?

File: validator/src/main/java/com/google/daq/mqtt/sequencer/sequences/BlobsetSequences.java [hunk 3]
```diff
@@ -46,4 +64,26 @@ public void endpoint_config_connection_error() {
     });
   }
 
+  @Test
+  @Description("Push endpoint config message to device that results in success.")
+  public void endpoint_config_connection_success() {
+    BlobBlobsetConfig config = new BlobBlobsetConfig();
+    config.phase = BlobPhase.FINAL;
+    config.base64 = endpointConfigBase64PayloadHostname(ENDPOINT_CONFIG_HOSTNAME);
```
Comment (grafnu) [inline validator/src/main/java/com/google/daq/mqtt/sequencer/sequences/BlobsetSequences.java:72]: method name seems off -- should the "Hostname" be there?

Comment (johnrandolph) [reply validator/src/main/java/com/google/daq/mqtt/sequencer/sequences/BlobsetSequences.java:72]: Right, for a while I had a different function that generated a payload with client_id. But now, it's removed. So I also renamed the function.

File: schema/configuration_endpoint.json (hunk 2 above)
Comment (johnrandolph) [reply schema/configuration_endpoint.json]: Copied from blob "nonce" regex.

**Review by grafnu — COMMENTED** [review]
Code looks ok, is there a passing CI sequencer run of this somewhere?

**Pull request merged**